oci-unit-tests/helper/test_helper.sh

```diff
@@ -15,15 +15,51 @@ load_shunit2() {
 
 debug() {
     if [ -n "${DEBUG_TESTS}" ]; then
-        if [ "${1}" = "-n" ]; then
-            shift
-            echo -n "$@"
-        else
-            echo "$@"
-        fi
+	if [ "${1}" = "-n" ]; then
+	    shift
+	    echo -n "$@"
+	else
+	    echo "$@"
+	fi
     fi
 }
 
+# Creates a new docker container from $DOCKER_IMAGE.
+#
+# By default, creates a temporary container running the software under
+# test as a daemon service.  The container is automatically removed
+# when the daemon exits.  All parameters to run_container_server() are
+# passed directly through to the contained software daemon.  The docker
+# behavior can be adjusted through environmental variables listed
+# below.  In particular, $suffix can be used to distinguish between
+# multiple containers used in a given test, such as 'client' and
+# 'server'.
+#
+# Parameters:
+#   $@: parameters to pass through to the container
+# Environment:
+#   $DOCKER_IMAGE: Name of the image containing the software to be tested.
+#       Defaults to "docker.io/ubuntu/<package>:edge"
+#   $DOCKER_PREFIX: Common prefix for all containers for this test.
+#       Defaults to "oci_<package>_test".
+#   $DOCKER_NETWORK: User-created network to connect the container to.
+#       Defaults to "oci_<package>_test_net".
+#   $SUFFIX: Optional tag for ensuring unique container names.
+#       Defaults to a random string.
+#
+# Stdout: Name of created container.
+# Returns: Error code from docker, or 0 on success.
+run_container_server() {
```

Comment:
I think many of our existing tests can benefit from this generic function (not all of them, though).  WDYT about search-and-replacing this function in our tests?  I think doing this in another PR would be better.

Reply:
Completely agree on both points, and will try to send a PR next week.